src/components/Channel/Channel.js

```diff
@@ -1,148 +1,674 @@
+    parent,
+    text,
+    ...extraFields
+  }) => {
+    channel.state.filterErrorMessages();
+
+    const messagePreview = createMessagePreview({
+      attachments,
+      extraFields,
+      mentioned_users,
+      parent,
+      text,
+    });
+
+    updateMessage(messagePreview, {
+      commands: [],
+      messageInput: '',
+      userAutocomplete: [],
+    });
+
+    await sendMessageRequest(messagePreview);
+  };
+
+  const retrySendMessage = async (message) => {
+    message = message.asMutable();
+    message.status = 'sending';
+    updateMessage(message);
+    await sendMessageRequest(message);
+  };
+
+  const loadMoreFinished = (updatedHasMore, messages) => {
+    if (unmounted) return;
+    setLoadingMore(false);
+    setHasMore(updatedHasMore);
+    setMessages(messages);
+  };
+
+  // hard limit to prevent you from scrolling faster than 1 page per 2 seconds
+  const loadMoreFinishedDebounced = debounce(loadMoreFinished, 2000, {
+    leading: true,
+    trailing: true,
+  });
+
+  const loadMore = async () => {
+    if (loadingMore || hasMore === false || unmounted) return;
+    setLoadingMore(true);
+
+    if (!messages.length) {
+      return setLoadingMore(false);
+    }
+
+    const oldestMessage = messages[0] ? messages[0] : null;
+
+    if (oldestMessage?.status !== 'received') {
+      return setLoadingMore(false);
+    }
+
+    const oldestID = oldestMessage ? oldestMessage.id : null;
+    const limit = 100;
+
+    try {
+      logger('Channel Component', 'Re-querying the messages', {
+        props,
+        limit,
+        id_lt: oldestID,
+      });
+
+      const queryResponse = await channel.query({
+        messages: { limit, id_lt: oldestID },
+      });
+
+      const updatedHasMore = queryResponse.messages.length === limit;
+      loadMoreFinishedDebounced(updatedHasMore, channel.state.messages);
+    } catch (e) {
+      console.warn('Message pagination request failed with error', e);
+      if (unmounted) return;
+      return setLoadingMore(false);
+    }
+  };
+
+  const loadMoreThrottled = throttle(loadMore, 2000, {
+    leading: true,
+    trailing: true,
+  });
+
+  // eslint-disable-next-line require-await
+  const editMessage = async (updatedMessage) => {
+    if (props.doUpdateMessageRequest) {
+      return Promise.resolve(
+        props.doUpdateMessageRequest(channel.cid, updatedMessage),
+      );
+    }
+    return client.updateMessage(updatedMessage);
+  };
+
+  const setEditingState = (message) => {
+    if (unmounted) return;
+    setEditing(message);
+  };
+
+  const clearEditingState = () => {
+    if (unmounted) return;
+    setEditing(false);
+  };
+
+  const removeMessage = (message) => {
+    channel.state.removeMessage(message);
+    if (unmounted) return;
+    setMessages(channel.state.messages);
+  };
+
+  /**
+   * THREAD METHODS
+   */
+
+  const openThread = (message) => {
+    const threadMessages = channel.state.threads[message.id] || [];
+    if (unmounted) return;
```

Comment:
```suggestion
    if (unmounted) return;
    const threadMessages = channel.state.threads[message.id] || [];
```

Reply:
kept previous order, so not sure about this change, erred on the side of not adjusting things i didn't have full context on